modDbImpl/src/org/aion/db/impl/leveldb/LevelDB.java

```diff
@@ -156,19 +156,49 @@ public boolean open() {
             }
 
             try {
+                LOG.warn("attempting to repair database {}", this.toString());
                 // attempt repair
                 JniDBFactory.factory.repair(f, options);
             } catch (Exception e2) {
                 LOG.error("Failed to repair the database " + this.toString() + " due to: ", e2);
+                // the repair failed
+                // close the connection and cleanup if needed
+                close();
             }
 
-            // close the connection and cleanup if needed
-            close();
+            // the repair didn't throw an exception
+            // try to open again
+            try {
+                db = JniDBFactory.factory.open(f, options);
+            } catch (Exception e2) {
+                LOG.error("Failed second attempt to open the database " + this.toString() + " due to: ", e2);
+                // close the connection and cleanup if needed
+                close();
```

Comment:
Minor: Do we have an established practice regarding "this". Can we pick between "close()" and "this.close()", and uniformly use either?

Reply:
I think it's dependent on the coder's preference.